src/pages/trending.tsx

```diff
@@ -0,0 +1,328 @@
+  Keyboard,
+  Cpu,
+  Layers,
+  LucideBellElectric,
+} from "lucide-react"
+import { Input } from "@/components/ui/input"
+import { Badge } from "@/components/ui/badge"
+import { SnippetTypeIcon } from "@/components/SnippetTypeIcon"
+import { timeAgo } from "@/lib/utils/timeAgo"
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select"
+
+const TrendingPage: React.FC = () => {
+  const axios = useAxios()
+  const apiBaseUrl = useSnippetsBaseApiUrl()
+  const [searchQuery, setSearchQuery] = useState("")
+  const [category, setCategory] = useState("all")
+
+  const {
+    data: snippets,
+    isLoading,
+    error,
+  } = useQuery<Snippet[]>("trendingSnippets", async () => {
+    const response = await axios.get("/snippets/list_trending")
+    return response.data.snippets
+  })
+
+  const filteredSnippets = snippets?.filter((snippet) => {
+    if (!searchQuery) return true
+
+    const query = searchQuery.toLowerCase().trim()
+
+    const searchableFields = [
+      snippet.unscoped_name.toLowerCase(),
+      snippet.owner_name.toLowerCase(),
+      (snippet.description || "").toLowerCase(),
+    ]
+
+    return searchableFields.some((field) => {
+      const queryWords = query.split(/\s+/).filter((word) => word.length > 0)
+      return queryWords.every((word) => field.includes(word))
+    })
+  })
+
+  const filteredByTypeSnippets = filteredSnippets?.filter((snippet) => {
+    if (category !== "all") {
+      const description = (
+        (snippet.description || "") + ((snippet as any).ai_description || "")
+      ).toLowerCase()
+      const name = snippet.unscoped_name.toLowerCase()
+
+      switch (category) {
+        case "keyboard":
+          return (
+            description.includes("keyboard") ||
+            name.includes("keyboard") ||
+            description.includes("keycap") ||
+            name.includes("keycap") ||
+            description.includes("keyboards") ||
+            name.includes("keyboards")
+          )
+        case "microcontroller":
+          return (
+            description.includes("microcontroller") ||
+            name.includes("mcu") ||
+            description.includes("arduino") ||
+            description.includes("esp32") ||
+            description.includes("raspberry") ||
+            name.includes("arduino") ||
+            name.includes("esp32") ||
+            name.includes("raspberry")
+          )
+        case "connector":
+          return (
+            description.includes("connector") ||
+            name.includes("connector") ||
+            description.includes("jack") ||
+            name.includes("jack") ||
+            description.includes("socket") ||
+            name.includes("socket")
+          )
+        case "sensor":
+          return (
+            description.includes("sensor") ||
+            name.includes("sensor") ||
+            description.includes("detector") ||
+            name.includes("detector")
```

Comment:
These should be done server side, can you add a parameter to list_trending, something like “tag=keyboard”